Frontend/src/app/resource-list-elem/resource-list-elem.component.spec.ts

```diff
@@ -1,23 +1,77 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
 import { ResourceListElemComponent } from './resource-list-elem.component';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { ResourceService } from '../resource.service';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { CommonModule } from '@angular/common';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { MatIconModule } from '@angular/material/icon';
+import { RouterModule } from '@angular/router';
+import { MatButtonModule } from '@angular/material/button';
+import { ActivatedRoute } from '@angular/router';
 
 describe('ResourceListElemComponent', () => {
   let component: ResourceListElemComponent;
   let fixture: ComponentFixture<ResourceListElemComponent>;
+  let resourceService: jasmine.SpyObj<ResourceService>;
+  let matDialog: jasmine.SpyObj<MatDialog>;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      imports: [ResourceListElemComponent]
-    })
-    .compileComponents();
+  beforeEach(waitForAsync(() => {
+    const resourceServiceSpy = jasmine.createSpyObj('ResourceService', ['deleteResource']);
+    const matDialogSpy = jasmine.createSpyObj('MatDialog', ['open']);
     
+    TestBed.configureTestingModule({
+      imports: [
+        ResourceListElemComponent,
+        MatDialogModule,
+        NoopAnimationsModule,
+        CommonModule,
+        MatIconModule,
+        MatExpansionModule,
+        RouterModule.forRoot([]),
+        MatButtonModule
+      ],
+      providers: [
+        { provide: ResourceService, useValue: resourceServiceSpy },
+        { provide: MatDialog, useValue: matDialogSpy },
+        { provide: ActivatedRoute, useValue: {} }
+      ],
+    }).compileComponents();
+
+    resourceService = TestBed.inject(ResourceService) as jasmine.SpyObj<ResourceService>;
```

Comment:
Nem használod sehol a resourceService változót, így eltávolítható innen és a 16. sorból is.

Reply:
javítva